File: mobile/lib/features/contest/domain/entities/contest_category.dart
import 'package:equatable/equatable.dart';

class ContestCategory extends Equatable {
  final String title;
  final String subject;
  final String contestId;
  final int numberOfQuestion;
  final String categoryId;
  final int userScore;
  final bool isSubmitted;
  const ContestCategory(
      {required this.title,
      required this.subject,
      required this.contestId,
      required this.numberOfQuestion,
      required this.categoryId,
      required this.isSubmitted,
      required this.userScore});
  @override
  List<Object?> get props => [
        title,
        subject,
        contestId,
      ];
}

File: new_backend/src/models/contestCategory.ts
import { Document, Schema, model } from 'mongoose';
import { Subject } from '../types/typeEnum';


export interface IContestCategoryDocument extends Document {
  title: string;
  subject: Subject;
  contestId: Schema.Types.ObjectId;
}

const ContestCategorySchema: Schema<IContestCategoryDocument> = new Schema(
  {
    title: {
      type: String,
      required: [true, 'Title is required'],
    },
    subject: {
      type: String,
      enum: Object.values(Subject),
      required: [true, 'Subject is required'],
    },
    contestId: {
      type: Schema.Types.ObjectId,
      ref: 'Contest',
      required: [true, 'Contest Id is required'],
    },
  },
  {
    timestamps: {
      createdAt: 'createdAt',
      updatedAt: 'updatedAt',
    },
  }
);

const ContestCategory = model<IContestCategoryDocument>('ContestCategory', ContestCategorySchema);

export default ContestCategory;
